### PR TITLE
Allow InfluxDB and Kapacitor configuration via ENV vars or CLI options

### PR DESCRIPTION
**Pull request by lukevmorris**
- [x] CHANGELOG.md updated
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #883 

## The Problem
Users want to be able to supply an `influxdb` and `kapacitor` configuration on `chronograf` startup.

## The Solution
Give the `chronograf` binary some more CLI options:
- `influxdb` stores the location of your InfluxDB instance
- `influxdb-username` stores your instance username
- `influxdb-password` stores your instance password
- `kapacitor` stores the location of your Kapacitor instance
- `kapacitor-username` stores your instance username
- `kapacitor-password` stores your instance password

### Behavior

Any instance provided via CLI will take precedence over instances already stored in your chronograf Bolt db -- they will act as default. If you restart the `chronograf` server without these configuration values, then the server will respect your previous default.

### Syntax
A user can now provide `influxdb` configuration as follows:
```
./chronograf --influxdb=http://localhost:8086 --influxdb-username=admin --influxdb-password=pass
```

Or as follows:
```
INFLUXDB=http://localhost:8086 INFLUXDB_USERNAME=admin INFLUXDB_PASSWORD=pass ./chronograf
```

And can provide `kapacitor` configuration as follows:
```
./chronograf --kapacitor=http://localhost:9092 --kapacitor-username=admin --kapacitor-password=pass
```

Or as follows:
```
KAPACITOR=http://localhost:9092 KAPACITOR_USERNAME=admin KAPACITOR_PASSWORD=pass ./chronograf
```

### CODE REVIEW

**Comment by jaredscheib** [comment]
Wow, nice. How about `influxdb-host` and `kapacitor-host` for the connection string?

File: server/server.go [hunk 1]
```diff
@@ -69,6 +78,115 @@ type Server struct {
 	handler           http.Handler
 }
 
+// LayoutBuilder is responsible for building Layouts
```
Comment (goller) [inline server/server.go]: I'd like to see all the builder interfaces and structs in a new file called builders.go

Comment (lukevmorris) [inline server/server.go]: you got it

Comment (lukevmorris) [inline server/server.go]: Addressed

**Comment by lukevmorris** [comment]
@goller What do you think of Jared's suggestion? Do we have any sort of naming convention already present in the app?

**Comment by goller** [comment]
@lukevmorris I think it should be INFLUXDB_URL/KAPACITOR_URL to emphasize that the parameter should include schema, host and port.

**Comment by lukevmorris** [comment]
@goller ok we'll go with that. Are you happy with the location of `builders.go`?

**Pull request merged**